code-of-conduct.md

```diff
@@ -1,3 +0,0 @@
-## Docker Distribution Community Code of Conduct
```

Comment:
I think GitHub uses this file in their "https://github.com/docker/distribution/community" rating, but it seems to have trouble detecting the file already, so not sure it's super important.

Reply:
Moved it into the `.github` directory. Since it is mentioned in the governance file, don't need it in the root. Also in a separate PR I think we should fill in the content of the code of conduct rather than just linking. While the CNCF CoC is a good template, it doesn't really apply to non-CNCF projects since the point of escalation is specifically for those projects.